src/cr/cube/matrix/assembler.py

```diff
@@ -117,14 +116,16 @@ def column_weighted_bases(self):
     @lazyproperty
     def columns_base(self):
         """1D/2D np.float64 ndarray of unweighted-N for each slice column/cell."""
-        # --- an MR_X slice produces a 2D column-base (each cell has its own N) ---
-        columns_base = self._measures.columns_base
-        if self._rows_dimension.dimension_type in (DT.MR_SUBVAR, DT.NUM_ARRAY):
-            return self._assemble_matrix(columns_base.blocks)
+        # --- an MR_X slice produces a 2D columns-base (each cell has its own N) ---
+        # --- This is really just another way to call the columns_weighted_bases ---
+        # --- TODO: Should column_base only be defined when it's 1D? This would
+        # --- require changes to exporter to use the bases to give a
+        # --- "column_base_range"
```

Comment:
Can't remember if I commented this on another PR, but yes, like rows-base and table-base, if columns-base is not 1D it is undefined and the caller can decide to use columns-base-range instead or grab the 2D column-unweighted-bases and figure out what it wants from there.

Whether that happens in this PR or a future one doesn't matter to me. Wherever is most convenient. It does probably make sense to do it in a PR by itself (or with closely related changes) so it's less of a big deal to coordinate with the `zoom` companion branch.